lib/wget.js

```diff
@@ -93,10 +93,20 @@ function download(src, output, options, _parentEvent, redirects) {
                 }
 
                 const flags = res.statusCode === 206 ? 'a' : 'w+';
-                writeStream = fs.createWriteStream(output, {
+                const writeStreamOptions = {
                     flags,
                     encoding: 'binary'
-                });
+                };
+
+                if (res.statusCode === 206 && options.start && options.total_size) {
+                    const buffer = new Buffer(options.total_size);
+                    buffer.fill('0');
+
+                    fs.writeFileSync(output, buffer, { flags: 'wx' });
```

Comment:
Is this really necessary? Any reference?

Reply:
Added in code comment.
> need to create buffer file whose size === total_size for createWriteStream's `start` option

https://stackoverflow.com/questions/16104406/specify-position-to-write-to-a-file-using-writable-stream-in-nodejs

Reply:
That seems counter-intuitive if that's the requirement. We would also not be able to create a file larger than 2G because of the buffer limitation with current code. (perhaps use https://nodejs.org/api/fs.html#fs_fs_ftruncate_fd_len_callback instead to avoid the buffer limit)

Were we encountering any problem by using createWriteStream directly?

Reply:
I think this would help us in this specific scenario
https://www.npmjs.com/package/fd-slicer

Reply:
After some tests, looks like an empty file is enough for createWriteStream.
Updated code to no longer use buffer, and it will now start from an empty file, then grow accordingly depending on the downloaded file.